### ISSUE TITLE
change in the result of polygon-line intersection from versions 3.8.1 to v3.9.0

### ISSUE THREAD

**Issue by davemfish** [libgeos/geos#408]
Apologies for not having a pure geos example. I observe a change in the result of a polygon-line intersection from geos versions 3.8.1 to 3.9.0. My shapely version is the same in both environments (1.7.1)

```
from shapely.geometry import Polygon, LineString
poly = Polygon(((-200, -200), (200, -200), (200, 200), (-200, 200), (-200, -200)))
line = LineString([(-100, 100), (-100, -100)])
print(poly.intersects(line))                       # always True
print(poly.intersection(line).wkt)                 # LINESTRING EMPTY in version 3.9.0
```
The intersection is `LINESTRING EMPTY` according to version 3.9.0, but is `LINESTRING (-100 100, -100 -100)` according to version 3.8.1.

I'm not entirely sure, is this a bugfix or a regression?

**Comment by pramsey** [comment]
It's a bug, gotta trust my lying eyes. Also confirm this behaviour is happening also on my copy of GEOS, which is state-of-the-art.

![geoms](https://user-images.githubusercontent.com/113883/107281114-1cc9b680-6a0e-11eb-91eb-36419822426c.png)

**Comment by davemfish** [comment]
Thanks for the confirmation. And thanks for GEOS!

**Comment by dr-jts** [comment]
Fixed in JTS.
Fixed in GEOS 3.10 ([e30955844d](https://git.osgeo.org/gitea/geos/geos/commit/e30955844dc6ad36669caee1c19619ebf8c54df4)).  
Backported to GEOS 3.9
